selfhost/typechecker.jakt

```diff
@@ -153,6 +153,10 @@ struct Typechecker {
     had_an_error: bool = false
     cpp_import_cache: [String:ScopeId] = [:]
     cpp_import_processor: CppImportProcessor? = None
+    temp_var_count: u64 = 0
+    // NOTE: This relies on Array being refcounted (shared).
```

Comment:
where?
I see the copy below (which won't actually copy the array), but that's not so much "relies" as it is a bug?

Reply:
oh, I'll be more explicit about that. I meant the mechanism of setting the current  block so statements can be added to it, I thought I explained it. Maybe putting the note on the paragraph

Reply:
Anyway, looks like I forgot to run the tests locally, so I'll move this to draft and mark when it's resolved again, thanks :)